Recycle Pod Template Check

pospispa opened the pull request:
The kube-controller-manager has two command line arguments (--pv-recycler-pod-template-filepath-hostpath and --pv-recycler-pod-template-filepath-nfs) that specify a recycle pod template. The recycle pod template may not contain the volume that shall be recycled.

A check is added to make sure that the recycle pod template contains at least a volume.

cc: @jsafrane

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36221)
<!-- Reviewable:end -->

ingvagabund commented:
@k8s-bot ok to test

pospispa commented:
Link to corresponding [Trello card](https://trello.com/c/WhCK1Ndv/86-5-add-checks-and-documentation-about-template-pods-for-recycling).

pmorie commented on cmd/kube-controller-manager/app/plugins.go:
```diff
@@ -24,9 +24,9 @@ import (
 	"fmt"
 
 	// Cloud providers
+
```
spurious newline, remove plz

pmorie commented on pkg/volume/plugins_test.go:
```diff
@@ -51,3 +51,71 @@ func TestSpecSourceConverters(t *testing.T) {
+func Test_ValidatePodTemplate(t *testing.T) {
+	timeout := int64(60)
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
```
you don't really need the ObjectMeta or most of the pod spec for these tests; you can trim down significantly here.

pmorie reviewed (commented):
Just a couple nits.

Commit:
Recycle Pod Template Check

The kube-controller-manager has two command line arguments (--pv-recycler-pod-template-filepath-hostpath and --pv-recycler-pod-template-filepath-nfs) that specify a recycle pod template. The recycle pod template may not contain the volume that shall be recycled.

A check is added to make sure that the recycle pod template contains at least a volume.

pospispa commented:
@pmorie your nits were addressed.

jsafrane commented:
/lgtm

jsafrane commented:
/lgtm

jsafrane commented:
> you can't LGTM a PR unless you are assigned as a reviewer.

I am reviewer!
